src/Design/03_InitialPoC_trans.purs

```diff
@@ -1,56 +1,52 @@
 module PSPS.Design.InitialPoC.Trans where
 
-import PSPS (class Bind, class Eval, App, kind Expr, kind ModuleName)
+import FFI.PSPS.Design.InitialPoC.Trans (class FFI_PSPS_Design_InitialPoC_Trans_Eval, Foreign'intAdd, Foreign'intMul)
+import PSPS (class Bind, class Eval, Abs, App, Foreign, kind Expr, kind ModuleName)
+import Type.Data.Peano (Succ, Z)
 
 foreign import data PSPS_Design_InitialPoC_Trans :: ModuleName
 
 class PSPS_Design_InitialPoC_Trans_Eval (expr :: Expr) (result :: Expr) | expr -> result
 
 class PSPS_Design_InitialPoC_Trans_Bind (path :: Symbol) (expr :: Expr) | path -> expr
 
-foreign import kind Maybe
-
-foreign import data Just :: Expr -> Maybe
-
-foreign import data Nothing :: Maybe
-
-foreign import data Maybe0 :: Maybe -> Expr
-
-foreign import data Maybe1 :: (Expr -> Maybe) -> Expr
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" (Maybe1 Just)
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Nothing :: PSPS_Design_InitialPoC_Trans_Bind "Nothing" (Maybe0 Nothing)
-
-instance iBind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" o => Bind PSPS_Design_InitialPoC_Trans "Just" o
+instance iPSPS_Design_InitialPoC_Trans_Bind'intAdd :: PSPS_Design_InitialPoC_Trans_Bind "intAdd" (Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd)))
 
-instance iBind'Nothing :: PSPS_Design_InitialPoC_Trans_Bind "Nothing" o => Bind PSPS_Design_InitialPoC_Trans "Nothing" o
+instance iPSPS_Design_InitialPoC_Trans_Eval'intAdd'intAdd1'App ::
+  PSPS_Design_InitialPoC_Trans_Eval a a' =>
+  PSPS_Design_InitialPoC_Trans_Eval (App (Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd))) a) (Abs "intAdd" Z (App (Foreign Foreign'intAdd) a'))
 
-instance iPSPS_Design_InitialPoC_Trans_Eval'Just'Maybe0 ::
-  PSPS_Design_InitialPoC_Trans_Eval v0 v0' =>
-  PSPS_Design_InitialPoC_Trans_Eval (Maybe0 (Just v0)) (Maybe0 (Just v0'))
+instance iPSPS_Design_InitialPoC_Trans_Eval'intAdd'intAdd0'App ::
```

Comment:
The same process continues: if we have the curried function over `a` from above applied to `b`, the compiler returns the foreign function being applied to both `a` and `b`.